applications/FluidDynamicsApplication/custom_elements/vectorial_convection_fractional_element.h

```diff
@@ -0,0 +1,310 @@
+
+    static constexpr unsigned int LocalSize = NumNodes * BlockSize;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constuctor.
+    /**
+     * @param NewId Index number of the new element (optional)
+     */
+    VectorialConvectionFractionalElement(IndexType NewId = 0);
+
+    /// Constructor using an array of nodes.
+    /**
+     * @param NewId Index of the new element
+     * @param ThisNodes An array containing the nodes of the new element
+     */
+    VectorialConvectionFractionalElement(IndexType NewId, const NodesArrayType &ThisNodes);
+
+    /// Constructor using a geometry object.
+    /**
+     * @param NewId Index of the new element
+     * @param pGeometry Pointer to a geometry object
+     */
+    VectorialConvectionFractionalElement(IndexType NewId, GeometryType::Pointer pGeometry);
+
+    /// Constuctor using geometry and properties.
+    /**
+     * @param NewId Index of the new element
+     * @param pGeometry Pointer to a geometry object
+     * @param pProperties Pointer to the element's properties
+     */
+    VectorialConvectionFractionalElement(IndexType NewId, GeometryType::Pointer pGeometry, Properties::Pointer pProperties);
+
+    /// Destructor.
+    virtual ~VectorialConvectionFractionalElement();
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    Element::Pointer Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const override;
+    Element::Pointer Create(IndexType NewId,GeometryType::Pointer pGeom,PropertiesType::Pointer pProperties) const override;
+
+    void CalculateLocalSystem(MatrixType& rLeftHandSideMatrix, VectorType& rRightHandSideVector, const ProcessInfo& rCurrentProcessInfo) override;
+    virtual void ComputeGaussPointLHSContribution(TElementData &rData,MatrixType &rLHS);
+    virtual void ComputeGaussPointRHSContribution(TElementData &rData,VectorType &rRHS);
+
+    void EquationIdVector(EquationIdVectorType& rResult, const ProcessInfo& rCurrentProcessInfo) const override;
+    void GetDofList(
+        DofsVectorType &rElementalDofList,
+        const ProcessInfo &rCurrentProcessInfo) const override;
+    void CalculateRightHandSide(
+        VectorType &rRightHandSideVector,
+        const ProcessInfo &rCurrentProcessInfo) override;
+    virtual void AddTimeIntegratedSystem(
+        TElementData &rData,
+        MatrixType &rLHS,
+        VectorType &rRHS);
+    virtual void UpdateIntegrationPointData(TElementData &rData,
+        unsigned int IntegrationPointIndex,
+        double Weight,
+        const typename TElementData::MatrixRowType &rN,
+        const typename TElementData::ShapeDerivativesType &rDN_DX) const;
+    virtual void CalculateGeometryData(Vector &rGaussWeights,
+                                       Matrix &rNContainer,
+                                       ShapeFunctionDerivativesArrayType &rDN_DX) const;
```

Comment:
This is a bit messy...

Reply:
Definition of each method has been done.